### DISCUSSION TITLE
How do you use `from` globally?

### DISCUSSION THREAD

**Discussion by CreativeTechGuy** [solidjs/solid#2126] [General]
Playground demo: https://playground.solidjs.com/anonymous/d1477536-d3f1-4ea0-b1fd-b2919db6993d

* Option 1 makes the most sense, but it results in the following warning being printed to the console: `​cleanups created outside a createRoot or render will never be run`
* Option 2 suppresses the warning by faking an `owner` object

What is the correct way to use `from` and `observable` so that it doesn't produce a warning (or whatever problems are indicated by the warning)?

It seems like the cleanup function will never run for a global signal anyway, so I cannot find any runtime impact of either option, both would work as expected. And the docs don't say anything about being forced to use `from` only within a component. Is there something I'm missing?

-----

Edit: It seems like another way to get an unowned block would be:

```js
let owner: Owner | null = null;
createRoot(() => {
    owner = getOwner();
});
// Later
runWithOwner(owner, () => from(global.valueObservable))
```
Not sure if that is better or worse than just constructing an owner object by hand.

**Comment by mdynnl** [comment]
with https://github.com/solidjs/solid/pull/1115, option 1 is fine now

**Reply by CreativeTechGuy** [reply]
Which release is that part of? My discussion post above was from many months after that linked PR was merged and it doesn't look like anything was changed in the last few days. So it doesn't seem like that worked.

**Reply by mdynnl** [reply]
ah, i mistook it with the usual disposal warning which the above PR was for.
the warning at issue is just doing its purpose as the disposal one. whatever method used to silence it doesn't really help there except the silencing part.

**Reply by CreativeTechGuy** [reply]
Okay so if we are intentionally using `from` globally and want to "acknowledge" the warning, we need to wrap it in a `runWithOwner`? That is the expected way to use this?

**Reply by mdynnl** [reply]
The warning actually comes from `onCleanup` only on development. If the intention is to silence the warning, yeah, you can use a) that, b) `createRoot(() => from(...))` , c) temporarily making console.warn noop. Either of them adds extra work to run though.

**Reply by CreativeTechGuy** [reply]
Yeah unfortunately there's no better way to disable that warning in certain places without adding too much more runtime code. It's not really something we can just ignore though since this is in a library we are creating and will be called in many places. We don't want our users to become desensitized to these errors and ignore them entirely since they may sometimes lead to meaningful issues.

It looks like performance-wise `runWithOwner` is the least impactful so that seems like the best thing we can use.
